Collector/MigrationsCollector.php

```diff
@@ -67,4 +72,26 @@ public function reset()
     {
         $this->data = [];
     }
+
+    private function filterUnavailableMigrations(
+        ExecutedMigrationsList $executedMigrations,
+        AvailableMigrationsList $availableMigrations
+    ): ExecutedMigrationsList {
+        $unavailableMigrations = array_filter($executedMigrations->getItems(), static function (ExecutedMigration $migration) use ($availableMigrations) {
+            return !$availableMigrations->hasMigration($migration->getVersion());
+        });
```

Comment:
Wouldn't these 3 lines be better in `ExecutedMigrationsList`?

```php
public function unavailableSubset(AvailableMigrationsList $availableMigrations): self
{
    return new self(array_filter($this->getItems(), static function (ExecutedMigration $migration) use ($availableMigrations)       {
        return !$availableMigrations->hasMigration($migration->getVersion());
    }));
}
```

Reply:
Good idea, thanks, done here: https://github.com/doctrine/migrations/pull/1174

Reply:
Ah I didn't realize that it was in a different package. @goetas , would you be fine with releasing 3.2.0 soon? Otherwise, we can merge this as is, and migrate to the new methods once 3.2.0 is out.